dotnet/src/SemanticKernel.Abstractions/Orchestration/FunctionResult.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.SemanticKernel.Orchestration;
+
+/// <summary>
+/// Function result after execution.
+/// </summary>
+public sealed class FunctionResult
```

Comment:
I am using `OpenAIFunctionResult` to indicate the function information returned by OpneAI when using the fuction calling feature. Do we need to better differentiate these with respect to naming?

Reply:
@gitri-ms How about `OpenAIFunctionCallResult`?

Reply:
Yeah, I thought about `FunctionExecutionResult` and `KernelExecutionResult` as well. I decided to make the name shorter.
I also thought about `FunctionResponse` and `KernelResponse`. But `Response` looks more closer to the networking communication, while `Result` should work better for code execution. By the way, what if you call your class `OpenAIFunctionCallResponse`?  

> as opposed to a result containing info about a function that has not been executed

It shouldn't be the case. We should have a result only when function is executed. In case of error, we should throw an exception.

Reply:
That's a fair point about Result vs. Response. I was actually using Response on my end but then changed to Result to better match ChatResult. But I think I'll go change it back. Thanks for talking this through with me!